src/psyclone/psyir/symbols/datatypes.py

```diff
@@ -291,6 +291,21 @@ def intrinsic(self):
             return self.partial_datatype.intrinsic
         return None
 
+    @property
+    def is_allocatable(self) -> bool:
+        '''If we have enough information in the partial_datatype,
+        determines if this data type is allocatable, and False otherwise.
```

Comment:
I think we have to be careful here to distinguish the case where we don't know from that where we do know that it's not allocatable. In other places we return `None` to indicate that we don't know something.

Reply:
Fair point, I return None now.